docs/js-core-1/week-2/lesson.md

```diff
@@ -382,253 +383,35 @@ Write a function, similar to the last exercise, that:
-
-```js
-const trainees = ["Ahmed", "Maria", "Atanas", "Nahidul", "Jack"];
-
-trainees[0]; // "Ahmed"
-trainees[3]; // "Nahidul"
-```
-
-You can also _assign_ new values to parts of an array:
-
-```js
-const trainees = ["Ahmed", "Maria", "Atanas", "Nahidul", "Jack"];
-
-trainees[2] = "Bianca";
-
-console.log(trainees); // ["Ahmed", "Maria", "Bianca", "Nahidul", "Jack"]
-```
-
-### Exercise (5 mins)
-
-:::note Exercise
-
-In Node, enter the following array:
-
-```sh
-> const fruits = ['banana', 'apple', 'strawberry', 'kiwi', 'fig', 'orange'];
-```
-
-Now, using the correct indexes, get the following values from the array:
-
-- strawberry
-- kiwi
-- orange
-- banana
-
-Then, replace 'apple' with 'raspberry', and replace 'fig' with 'pineapple'.
-
-:::
-
-### Exercise (5 mins)
-
-:::note Exercise
-
-Complete this function so that, if the second element in the array contains the name "Amy", it returns `"Second element matched!"`
-
-```js
-function secondMatchesAmy(array) {
-  if ( ) {
-    return "Second element matched!";
-  }
-  return "Second element not matched";
-}
-```
-
-:::
-
-### Using for loops with arrays
-
-We can use the power of loops to run some code for each element in our array.
-
-When we do this say we **iterate** over an array.
-
-```js
-const daysOfWeek = [
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-  "Sunday",
-];
-
-for (let i = 0; i < daysOfWeek.length; i++) {
-  const dayMessage = "day is: " + daysOfWeek[i];
-  const indexMessage = "index is: " + i;
-  console.log(indexMessage, dayMessage);
-}
-```
-
-### Exercise (10 mins)
-
-:::note Exercise
-
-Write a function which takes your **trainees** array as an input. In the function, use a for loop to **iterate** over the array and print the name of each trainee to the console.
-
-:::
-
-### Exercise - extra credit (20 mins)
-
-:::note Exercise
-
-In pairs, write a function which squares all numbers in an array and returns the array of squared numbers.
-
-Write a second function which takes an input of arrays and only returns an array of even numbers.
-
-How can you combine the two functions to return an array of even and squared numbers?
+- You should use a for loop
 
 :::
 
-## Extra exercises
-
-:::note Exercise
-
-### Playing computer I
-
-1. Working in pairs or groups, you have to predict the output of this program without executing it.
-2. What is printed to the console?
-3. Have you learned anything new during this exercise?
-
-```js
-const daysOfWeek = [
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-  "Sunday",
-];
-
-function workingDay(day) {
-  return day + " is a working day";
-}
-
-function weekendDay(day) {
-  return day + " is at the weekend!";
-}
+### for of loop
```

Comment:
I think this section got mangled? Also, maybe we should move this to the arrays section? It's probably easier to explain with arrays than strings? But 🤷 

Either way, it's currently a little broken, though!

Reply:
resolved by hastily writing https://github.com/CodeYourFuture/syllabus/pull/458/commits/40f5170e6cfda5361840b6009943e9ee9e288b14